src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -258,28 +259,88 @@ def __len__(self) -> int:
         """
         return len(self._buffer)
 
-    @overload
-    def __getitem__(self, key: SupportsIndex) -> float:
-        """See the main __getitem__ method.
+    def window(
+        self,
+        start: int | datetime,
+        end: int | datetime,
+        fill_gaps: Optional[float] = None,
+        force_copy: bool = False,
+    ) -> ArrayLike:
+        """
+        Access a sub window of the `MovingWindow`.
 
-        # noqa: DAR101 key
+        Args:
+            start: The start index or datetime of the sub window.
+            end: The end index or datetime of the sub window.
+            fill_gaps: The value to fill gaps with, fill_gaps implies force_copy = True.
+            force_copy: If True, a copy of the underlying buffer is returned.
```

Comment:
I would add a note here that usually the data is copied even if `False` if the requested window size is of similar size than the buffer.